src/main/java/at/codersbay/datastructures/Fractional.java

```diff
@@ -5,70 +5,133 @@
  * e.g.: 1/2, 1/3, 24/37, ..
  */
 public class Fractional {
+	
+	private Integer numerator;
+	private Integer denominator;
 
     public Fractional(Integer numerator, Integer denominator) {
-        // TODO implement to fix tests in FractionalTest1
+    	/*numerator can be 0 but not null*/
+    	if(numerator == null) {
+    		throw new IllegalArgumentException();
```

Comment:
IIRC the test requires the exception to have a specific message ;) you sure the tests pass with this?

Reply:
Tests passed,  didn't mind a missing or false message.
I'm not to firm with the Testing annotation & syntax,
maybe needs an assertEquals for the message following the assertThrows?
https://www.journaldev.com/21826/junit-assert-exception-expected